### PR TITLE
Compose enhancements

### PR DESCRIPTION
**Pull request by ooogway**
* Compose Enhancements:
   * `$resolve` for mapping services to context [(relevant code)](https://github.com/ooogway/chairo/blob/master/lib/index.js#L126-L139) 
   * Hoek templates for nested objects in context [(relevant code)](https://github.com/ooogway/chairo/blob/master/lib/index.js#L132-L135)
* Moved Defaults and Schema to separate files
* Default Seneca configuration: [(relevant code)](https://github.com/ooogway/chairo/blob/master/lib/defaults.js#L17-L31)
  * log: silent
  * actcache: disabled
  * default plugins: none
  * web: false
* README update

@geek: need your views on:

1. default seneca configuration: all tests pass with stripped down version of seneca. However, a cleaner way to enable plugins would be desirable, something like `enableWeb`: `boolean` OR `function`, or perhaps something like glue ? 
2. Hoek templates for $resolve object in compose: I haven't really thought thru the edge cases where this can go wrong
3. is it a good idea to register vision as a part of Chairo initialization ?

### CODE REVIEW

File: lib/defaults.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// Load Modules
+
+const Jsonic = require('jsonic');
+const Hoek = require('hoek');
+
+// Internals
+
+const internals = {};
+
+exports.apply = function (type, options) {
```
Comment (geek) [inline lib/defaults.js]: thoughts on just exposing seneca and cache as functions.  so its `Defaults.seneca(options);` and `Defaults.cache(options);` since we only have 2 defaults

File: lib/schema.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+// Declare internals
+
+const internals = {};
+
+exports.apply = function (type, options, message) {
+
+    const result = Joi.validate(options, internals[type]);
+    Hoek.assert(!result.error, 'Invalid', type, 'options', message ? '(' + message + ')' : '', result.error && result.error.annotate());
+    return result.value;
```
Comment (geek) [inline lib/schema.js]: instead of going through apply, thoughts on `Schema.action(options, message);` ?

**Comment by geek** [comment]
@ooogway awesome, thanks for the contribution

1. do you mean from within chairo?
2. feel free to add more tests around this to try and catch any edge cases
3. I think we should leave vision out of the registration, since we want to let people create their own template mappings processing, otherwise we will end up cluttering our configuration in chairo with vision options.

**Comment by ooogway** [comment]
@geek regarding Schema and Defaults apply methods, I had similar thoughts since we have limited number of schemas and defaults. I chose to keep it this way for future where we might add seneca hapi plugin inter-op or other features which might lead to more schemas and defaults. That said, we can change it when it comes to it. I will make changes in current PR as advised by you.

Regarding #1 (seneca default plugins config from chairo) I brought this up after looking at how web plugin was handled - perhaps making seneca options a little easier to handle.

#2 will add more tests

#3 sounds good

**Comment by geek** [comment]
@ooogway we are working on improving how options are passed around in seneca.  Thanks for the update.  I will publish once ready.
